### ISSUE TITLE
Question: Are there plans to support QCC as compiler for QNX platform?

### ISSUE THREAD

**Issue by TobiasSchwindl** [conan-io/cmake-conan#438]
Hi there,

currently the target "_conan_detect_compiler" does not support the qcc compiler. Are there any plans to support it? 
Or what would be needed in order to support the qcc compiler for QNX?

qcc is heaviliy based on the gcc compiler, but contains adaptions for the QNX platform. Any hints/ideas or as asked previously plans to support the qcc compiler out of the box?

Thanks for any help/input you can provide.

**Comment by czoido** [comment]
Hi @virstl,
`_conan_detect_compiler` is just a a macro to detect the most typical compilers, if you have specific needs that are not covered by the autodetection of the compiler you can pass that setting to `conan_cmake_install` like:

```
conan_cmake_install(...
                    SETTINGS compiler=qcc
                    SETTINGS compiler.version=5.4)
```

You can also pass a profile and just skip the whole autodetection.

Hope this helps!

**Comment by TobiasSchwindl** [comment]
Hi @czoido ,

thanks for the fast response. If I understood you correctly you dont want to support the qcc compiler within the auto-detection. With the mentioned solutions I think I understand the reasons to do so...

And also thanks for clarification - I will try out your suggestion to overwrite some needed settings for qcc. I am not sure if I want to give up all the "auto-detection", but maybe that depends on my success to pass relevant arguments to "conan_cmake_install".

Thanks for your help regarding this topic.

**Comment by TobiasSchwindl** [comment]
After trying a few things I out:
Is something like this https://github.com/conan-io/cmake-conan/compare/develop...virstl:cmake-conan:change-support-qcc needed to get it going with the qcc compiler anway? Even with your mentioned solutions I always failed since this compiler detection macro seems always to be executed somehow (but I am not a conan nor a cmake expert, just seems to be the case for me here)

Any chance that such a change could be included? (Not necessarily the exact change, but something like it)

**Comment by czoido** [comment]
Hi @virstl,
I can't see QCC as a valid value for [CMAKE_<LANG>_COMPILER_ID](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html).
Let me do some further investigation about what we can do in the cmake-conan side and I'll also do some tests and check why the workaround I suggested is not working and ping you back.

**Comment by czoido** [comment]
I have [opened this PR](https://github.com/conan-io/cmake-conan/pull/439) to check, we will consider if it can be merged.
Looks like although the `QCC` compiler id does not appear in the [docs](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html), its mentioned in this policy: https://cmake.org/cmake/help/latest/policy/CMP0047.html

@virstl, could you please test the PR?

**Comment by TobiasSchwindl** [comment]
Hi @czoido ,

thanks for your continued support here. The link you provided is also where I found out that QCC is known in cmake since 3.0. I am not sure if you support older versions of cmake to additionally include a workaround for older versions?

Anyway, the fix you provided works like a charm

**Comment by czoido** [comment]
Hi @virstl,
We have just merged https://github.com/conan-io/cmake-conan/pull/439 and will be included in the 0.19 release
cmake-conan 0.19 will include support for qcc
